src/app/services/hiscores/hiscore.service.ts

```diff
@@ -0,0 +1,136 @@
+import { Injectable } from '@angular/core';
+import { HiscoreEntry } from '@osrs-tracker/models';
+import partition from 'lodash-es/partition';
+import {
+  BossEnum,
+  Hiscore,
+  MiniGame,
+  MiniGameEnum,
+  ParseOrder,
+  PO_2023_03_15,
+  RaidEnum,
+  Skill,
+  SkillEnum,
+} from './hiscore.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class HiscoreService {
+  parseHiscores(hiscoreEntries: HiscoreEntry[]): Hiscore[] {
+    return hiscoreEntries.map(
+      hiscoreEntry =>
+        ({
+          ...hiscoreEntry,
+          ...this.parseHiscoreString(hiscoreEntry.sourceString),
+        } as Hiscore),
+    );
+  }
+
+  parseHiscoreString(hiscoreString: string): Omit<Hiscore, keyof HiscoreEntry> {
+    const lines = hiscoreString.split('\n').filter(line => line.length);
+
+    const skills = lines.slice(0, 24).map((line, i) => this.parseSkillLine(PO_2023_03_15, line, i));
+    const leaguePoints = lines.slice(24, 25).map((line, i) => this.parseMiniGameLine(PO_2023_03_15, line, 24 + i))[0];
+    const bountyHunter = lines.slice(25, 27).map((line, i) => this.parseMiniGameLine(PO_2023_03_15, line, 25 + i));
+    const clueScrolls = lines.slice(27, 35).map((line, i) => this.parseMiniGameLine(PO_2023_03_15, line, 27 + i));
+    const miniGames = lines.slice(34, 38).map((line, i) => this.parseMiniGameLine(PO_2023_03_15, line, 34 + i));
+    const bossesAndRaids = lines.slice(38).map((line, i) => this.parseMiniGameLine(PO_2023_03_15, line, 38 + i));
```

Comment:
Probably should add those ranges in the `PO_2023_03_15` as wel.

ex.
```ts
export type ParseOrder = {
  ranges: {
    skills: [number, number];
    leaguePoints: [number, number];
    bountyHunter: [number, number];
    clueScrolls: [number, number];
    miniGames: [number, number];
    bossesAndRaids: [number, number];
  };
  individualOrder: (BossEnum | MiniGameEnum | RaidEnum | SkillEnum)[];
}
```

Reply:
Or maybe only keep the skill count, and everything after that is `parseMiniGameLine`, and then partition them.

Or maybe just filter the skills from the parse order, and then? This needs some thinking to find a nice future proof way.

Reply:
done